schema/elements.go

```diff
@@ -117,6 +122,20 @@ type Map struct {
 	// The default behavior for maps is `separable`; it's permitted to
 	// leave this unset to get the default behavior.
 	ElementRelationship ElementRelationship `yaml:"elementRelationship,omitempty"`
+
+	once sync.Once
+	m    map[string]TypeRef
+}
+
+func (m *Map) FindType(name string) (TypeRef, bool) {
```

Comment:
Maps have fields, not types?

Reply:
the name is probably not the best. We're looking for the type for a specific field

Reply:
`FindFieldType`?

Reply:
No, FindField--callers can get the type from the result.